src/js/actions/tools.js

```diff
@@ -338,8 +375,15 @@ define(function (require, exports) {
 
     var resetSuperselect = {
         command: resetSuperselectCommand,
-        reads: [locks.JS_APP, locks.JS_TOOL, locks.JS_SHORTCUT],
-        writes: [locks.PS_APP, locks.JS_POLICY, locks.PS_TOOL, locks.JS_TOOL, locks.JS_SHORTCUT]
+        reads: [locks.JS_APP, locks.JS_TOOL, locks.JS_SHORTCUT, locks.PS_DOC, locks.JS_DOC],
+        writes: [locks.PS_APP, locks.JS_POLICY, locks.PS_TOOL, locks.JS_TOOL, locks.JS_SHORTCUT,
+            locks.PS_DOC, locks.JS_DOC]
```

Comment:
note to self: sweet glory, this is a lot of locks.  if all these write locks are really needed, then maybe that means this action has too much going on.  Or maybe this just really is the intersection of a lot of moving parts.  cc @iwehrman